application/ingest_change_event/ingest_change_event.py

```diff
@@ -0,0 +1,103 @@
+from json import dumps
+from os import environ
+from time import gmtime, strftime, time_ns
+
+from aws_embedded_metrics import metric_scope
+from aws_lambda_powertools.logging import Logger
+from aws_lambda_powertools.tracing import Tracer
+from aws_lambda_powertools.utilities.data_classes import event_source, SQSEvent
+from aws_lambda_powertools.utilities.typing.lambda_context import LambdaContext
+from boto3 import client
+
+from .change_event_validation import validate_change_event
+from common.dynamodb import add_change_event_to_dynamodb, get_latest_sequence_id_for_a_given_odscode_from_dynamodb
+from common.middlewares import set_correlation_id, unhandled_exception_logging
+from common.types import HoldingQueueChangeEventItem
+from common.utilities import extract_body, get_sequence_number, remove_given_keys_from_dict_by_msg_limit
+
+logger = Logger()
+tracer = Tracer()
+sqs = client("sqs")
+
+
+@unhandled_exception_logging()
+@tracer.capture_lambda_handler()
+@event_source(data_class=SQSEvent)
+@logger.inject_lambda_context(clear_state=True)
+@set_correlation_id()
+@metric_scope
+def lambda_handler(event: SQSEvent, context: LambdaContext, metrics) -> None:
+    """Entrypoint handler for the ingest change event lambda
+
+    This lambda runs the change event validation, puts the change event on the dynamodb table
+    and then sends the validated change event to the delay queue.
+
+    Args:
+        event (SQSEvent): Lambda function invocation event
+        context (LambdaContext): Lambda function context object
+
+    Event: The event payload should contain an Update Request
+    """
+    time_start_ns = time_ns()
+    if len(list(event.records)) != 1:
+        raise ValueError(f"{len(list(event.records))} records found in event. Expected 1.")
+
+    record = next(event.records)
+    change_event = extract_body(record.body)
+    validate_change_event(change_event)
```

Comment:
what happen if the calling function raise exception?

Reply:
It's logged and then we give up processing the change event if it doesn't contain the correct org type or org sub type. I forget it tries this process, before sending the message to the change event dlq handler to make sure the message is archived in dynamodb